src/k2/httpproxy/HTTPProxy.cpp

```diff
@@ -316,7 +234,38 @@ HTTPProxy::_handleRead(shd::ReadRequest&& request) {
 seastar::future<std::tuple<sh::Status, shd::QueryResponse>>
 HTTPProxy::_handleQuery(shd::QueryRequest&& request) {
     K2LOG_D(log::httpproxy, "Received query request {}", request);
-    return MakeHTTPResponse<shd::QueryResponse>(sh::Statuses::S501_Not_Implemented("query not implemented"), shd::QueryResponse{});
+    auto iter = _txns.find(request.timestamp);
+    if (iter == _txns.end()) {
+        return MakeHTTPResponse<shd::QueryResponse>(Txn_S410_Gone, shd::QueryResponse{});
+    }
+    auto queryIter = iter->second.queries.find(request.queryId);
+    if (queryIter ==iter->second.queries.end()) {
+        return MakeHTTPResponse<shd::QueryResponse>(sh::Statuses::S410_Gone("Query does not exist"), shd::QueryResponse{});
```

Comment:
We should log a warning here - it signals a bad client state and should not happen since we're the ones keeping track of transactions.